midend/parserUnroll.cpp

```diff
@@ -23,75 +24,32 @@ bool StackVariable::repOk(const IR::Expression* expr) {
     return member->expr->is<IR::PathExpression>() || repOk(member->expr);
 }
 
-bool StackVariable::operator<(const StackVariable& other) const {
-    return compare(member, other.member) < 0;
-}
-
 bool StackVariable::operator==(const StackVariable& other) const {
     // Delegate to IR's notion of equality.
     return *member == *other.member;
 }
 
-int StackVariable::compare(const IR::Expression* e1, const IR::Expression* e2) {
-    if (const auto* m1 = e1->to<IR::Member>()) {
-        return compare(m1, e2);
-    }
-    if (const auto* p1 = e1->to<IR::PathExpression>()) {
-        return compare(p1, e2);
-    }
-    BUG("Not a valid StateVariable: %1%", e1);
-}
-
-int StackVariable::compare(const IR::Member* m1, const IR::Expression* e2) {
-    if (const auto* m2 = e2->to<IR::Member>()) {
-        return compare(m1, m2);
-    }
-    if (e2->is<IR::PathExpression>()) {
-        return 1;
-    }
-    BUG("Not a valid StateVariable: %1%", e2);
-}
-
-int StackVariable::compare(const IR::Member* m1, const IR::Member* m2) {
-    auto result = compare(m1->expr, m2->expr);
-    if (result != 0) {
-        return result;
-    }
-    if (m1->member.name < m2->member.name) {
-        return -1;
-    }
-    if (m1->member.name > m2->member.name) {
-        return 1;
-    }
-    return 0;
-}
-
-int StackVariable::compare(const IR::PathExpression* p1, const IR::Expression* e2) {
-    if (const auto* p2 = e2->to<IR::PathExpression>()) {
-        return compare(p1, p2);
-    }
-    if (e2->is<IR::Member>()) {
-        return -1;
-    }
-    BUG("Not a valid StateVariable: %1%", e2);
-}
-
-int StackVariable::compare(const IR::PathExpression* p1, const IR::PathExpression* p2) {
-    if (p1->path->name.name < p2->path->name.name) {
-        return -1;
-    }
-    if (p1->path->name.name > p2->path->name.name) {
-        return 1;
+size_t StackVariableHash::operator()(const StackVariable& var) const {
+    const IR::Member* curMember = var.operator->();
+    std::vector<size_t> h;
+    while(curMember) {
```

Comment:
missing space

Reply:
This needs to be fixed, then I can merge @VolodymyrPeschanenkoIntel.